index.js

```diff
@@ -1,12 +1,34 @@
-import errorEx from 'error-ex';
 import fallback from 'json-parse-even-better-errors';
 import {codeFrameColumns} from '@babel/code-frame';
 import {LinesAndColumns} from 'lines-and-columns';
 
-export const JSONError = errorEx('JSONError', {
-	fileName: errorEx.append('in %s'),
-	codeFrame: errorEx.append('\n\n%s\n'),
-});
+export class JSONError extends Error {
+	constructor(message) {
+		super(message);
+
+		let _messages = message instanceof Error
+			? message.message
+			: message;
+
+		delete this.message;
+
```

Comment:
You may need:

```js
if (Error.captureStackTrace) {
  Error.captureStackTrace(this, JSONError);
}
```

Reply:
Please, take a look at the [documentation](https://nodejs.org/docs/latest-v8.x/api/errors.html#errors_new_error_message). The most important part:

> The error.stack property will represent the point in the code at which new Error() was called. Stack traces are dependent on V8's stack trace API. Stack traces extend only to either (a) the beginning of synchronous code execution, or (b) the number of frames given by the property Error.stackTraceLimit, whichever is smaller.

.stack will be added automatically.

But to be safe, i will add it